### PR TITLE
DB-8896 Introduce DB2 compatible parameter to csv read "" as empty string 

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

File: splice_machine/src/main/java/com/splicemachine/derby/impl/storage/SpliceRegionAdmin.java [hunk 1]
```diff
@@ -775,14 +778,21 @@ public static  ConglomerateDescriptor getIndex(TableDescriptor td, String indexN
...
         // set up csv reader
         CsvPreference preference = createCsvPreference(columnDelimiter, characterDelimiter);
         Reader reader = new StringReader(splitKey);
         List<Integer> valueSizeHints = new ArrayList<>(execRow.nColumns());
         for(DataValueDescriptor dvd : execRow.getRowArray()) {
             valueSizeHints.add(dvd.estimateMemoryUsage());
         }
-        MutableCSVTokenizer tokenizer = new MutableCSVTokenizer(reader,preference, false,
+
+        boolean quotedEmptyIsNull = !PropertyUtil.getCachedDatabaseBoolean(
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/storage/SpliceRegionAdmin.java:793]: I think our current behaviour is wrong (treating quoted empty strings as null). Shouldn't our default behaviour be the "DB2-compatible" one?

**Review by dgomezferro — CHANGES_REQUESTED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/storage/SpliceRegionAdmin.java (hunk 1 above)
Comment (dgomezferro) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/storage/SpliceRegionAdmin.java:793]: @arnaud-splice brought me up to speed regarding yesterday's discussion.

**Review by dgomezferro — APPROVED** [review]

**Review by martinrupp — APPROVED** [review]

**Review by jyuanca — APPROVED** [review]

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**